modules/widgets/blogs-i-follow/blogs-i-follow.php

```diff
@@ -0,0 +1,540 @@
+			$response_body = json_decode( $response_body );
+			$followed_blogs = array_map( array('Jetpack_Widget_Blogs_I_Follow', 'convert_rest_subscription'), $response_body->subscriptions );
+			$followed_blogs = Jetpack_Widget_Blogs_I_Follow::populate_blog_details( $followed_blogs );
+			return $followed_blogs;
+		}
+	}
+
+	/**
+	 * get the subscriptions used by the widget
+	 * result is cached using a transient
+	 *
+	 * @return array the subscriptions
+	 */
+	function get_subscriptions() {
+		$this->subscriptions[$this->id] = get_transient( $this->id . '-subscriptions' );
+
+		// TODO: Remove DEBUG
+		if ( WP_DEBUG || empty( $this->subscriptions[$this->id] ) ) {
+			delete_transient( $this->id . '-widget' );
+
+			$this->subscriptions[$this->id] = array();
+			$this->subscriptions[$this->id]['user_id'] = $this->user_id;
+			$subscription_args = array( 'user_id' => $this->user_id, 'public_only' => true );
+			// TODO: For WordPress.com, hook into this filter and use wpcom_subs_get_blogs
+			// and other related functions to populate the subscription data
+			$this->subscriptions[$this->id]['subscriptions'] = apply_filters( 'jetpack_populate_blog_subscriptions', null, $subscription_args );
+
+			if ( is_array( $this->subscriptions[$this->id]['subscriptions'] ) ) {
+				foreach ( $this->subscriptions[$this->id]['subscriptions'] as &$sub ) {
+					if ( ! wp_startswith( $sub['blog_url'], 'http://' ) && ! wp_startswith( $sub['blog_url'], 'https://' ) ) {
+						$sub['blog_url'] = 'http://' . $sub['blog_url'];
+					}
+				}
+
+				if ( !empty( $this->subscriptions[$this->id]['subscriptions'] ) ) {
+					$this->subscriptions[$this->id]['subscriptions'] = array_slice( $this->subscriptions[$this->id]['subscriptions'], 0, $this->number );
+					set_transient( $this->id . '-subscriptions', $this->subscriptions[$this->id], self::$expiration );
+				}
+			}
+		}
+
+		return $this->subscriptions[$this->id]['subscriptions'];
+	}
+
+	/**
+	 * determine if the widget should be hidden or shown
+	 *
+	 * @param array $subscriptions the subscriptions for the current widget
+	 * @return bool
+	 */
+	function shouldnt_show( $subscriptions ) {
+		return ( empty( $subscriptions ) && get_current_user_id() != $this->user_id );
+	}
+
+	/**
+	 * Infer the blog name from the subscription URL(s) when the name is not available
+	 *
+	 * @param array $subscription the subscription data lacking a blog name
+	 * @return string the inferred blog name
+	 */
+	function get_inferred_blog_name( $subscription ) {
+		return rtrim( str_replace( 'http://', '', empty( $subscription['blog_url'] ) ? $subscription['feed_url'] : $subscription['blog_url'] ), '/' );
```

Comment:
Good catch! I'll change this to consider both. The code I originally inherited used this line:

    $sub['blog_name'] = rtrim( str_replace( 'http://', '', empty( $sub['blog_url'] ) ? $sub['feed_url'] : $sub['blog_url'] ), '/' );